content/en/blog/2023/network-security-splunk/index.md

```diff
@@ -0,0 +1,70 @@
+
+{{< image width="90%"
+    link="Splunk Platform.png"
+    caption="Splunk Application Platform"
+    >}}
+
+## Pain Points
+
+While Splunk’s architecture meets many of our needs, there are a few pain points worth discussing.  As you recall, Istio operates by creating as many Envoy Sidecars as application pods, which is an inefficient use of resources.  In addition, when a particular application has unique needs from its sidecar, such as additional CPU or Memory, it can be difficult to adjust these settings without adjusting them for all sidecars in the mesh.  Istio Sidecar injection involves a lot of magic, using a mutating webhook to add a sidecar container to every pod as it is created, which means those pods no longer match their corresponding deployments.  Additionally, injection can only happen at pod creation time, which means that any time a sidecar version or parameter is updated, all pods must be restarted before they will get the new settings.  Overall, this magic complicates running a service mesh in production, and adds a great deal of operational uncertainty to your application.
+
+The Istio project is aware of these limitations, and believes they will be substantially improved by the new Ambient mode for Istio.  In this mode, Layer 4 constructs like identity and encryption will be applied by a Daemon running on the node, but not in the same pod as the application.  Layer 7 features will still be handled by Envoy, but Envoy will be run in an adjacent pod as part of its own deployment, rather than relying on the magic of sidecar injection.  Application pods will not be modified in any way in ambient mode, which should add a good deal of predictability to service mesh operations.  Ambient mode is expected to reach Alpha quality in Istio 1.18.
+
+## Conclusion
+
+With all these layers to network security at Splunk Cloud, it is helpful to take a step back and examine the life of a request as it traverses these layers.  When a client sends a request, they first connect to our NLB, which will be allowed or blocked by our `VPC ACLs`. The NLB then proxies the request to one of our ingress nodes, which terminates TLS and inspects the request at Layer 7, choosing to allow or block the request. The Envoy Gateway then validates the request using `ExtAuthZ` to ensure it is properly authenticated, and meets quota restrictions before being allowed into the cluster. Next, the Envoy Gateway proxies the request upstream, and the network policies from Kubernetes take effect again to make sure this proxying is allowed. The upstream sidecar on the workload inspects the Layer 7 requests and if allowed, it will decrypt the request and send it to the workload in clear text.
```

Comment:
```suggestion
With all these layers to network security at Splunk Cloud, it is helpful to take a step back and examine the life of a request as it traverses these layers.  When a client sends a request, they first connect to the NLB, which will be allowed or blocked by the `VPC ACLs`. The NLB then proxies the request to one of the ingress nodes, which terminates TLS and inspects the request at Layer 7, choosing to allow or block the request. The Envoy Gateway then validates the request using `ExtAuthZ` to ensure it is properly authenticated, and meets quota restrictions before being allowed into the cluster. Next, the Envoy Gateway proxies the request upstream, and the network policies from Kubernetes take effect again to make sure this proxying is allowed. The upstream sidecar on the workload inspects the Layer 7 requests and if allowed, it will decrypt the request and send it to the workload in clear text.
```